### PR TITLE
fix: sveltekit route conflicts

### PR DESCRIPTION
**Pull request by monsieurswag**
A `route conflict` occurs when a generic route containing a `+server.ts` file (e.g. `(app)/(internal)/[model=urlmodel]/[id=uuid]/+server.ts`) has its `+server.ts` request handlers hidden by a matching route (e.g. `(app)/(internal)/assets/[id=uuid]`) which contains a `+page.svelte` but does not contain a `+server.ts`.

The problem is that a the `+server.ts` of the generic route will not answer to request due to the existence of the `+page.svelte` in the matching route.

This PR:
- Adds a new test (in `frontend/src/lib/utils/api-routes-conflicts.test.ts‎`) to detect route conflicts.
- Add the missing `+server.ts` to fix the existing route conflicts in the codebase.
- Adds reusable generic request handlers in `frontend/src/lib/utils/api-routes.ts‎`.

**WARNING:** This test DOES NOT check for `frontend/enterprise/src/routes` route conflicts yet.

## Summary by CodeRabbit

* **New Features**
  * Centralized API request handlers and many new GET/PATCH endpoints wired for multiple resources

* **Bug Fixes**
  * Avoid appending empty query strings to downstream API requests

* **Tests**
  * Added a test validating route-handler consistency across the app

* **Refactor**
  * Consolidated duplicate request-handling into reusable utilities
  * Replaced per-call parameter scans with precomputed lookups; removed some debug logging

### CODE REVIEW

File: frontend/src/lib/utils/table.ts [hunk 1]
```diff
@@ -3491,3 +3490,17 @@ export const headData = (model: urlModel) =>
...
+		.filter((field) => field.hasOwnProperty("filters"))
+		.flatMap((field) => Object.keys(field.filters))
+);
+
+/** Example fields: "folder", "lc_status", "filtering_labels", etc. */
+export const fieldSet = new Set<string>(
+	Object.values(listViewFields)
+		.filter((field) => field.hasOwnProperty("body"))
+		.flatMap((field) => field.body)
+);
```
[the remark this answers is not in the document]
Comment (monsieurswag) [reply frontend/src/lib/utils/table.ts]: I will use single quotes.
I don't get why you complain about `filter.filters` though, yes there is no type checking after `field.hasOwnProperty("filters")` but filters SHALL always be objects, if they aren't the best behavior is to throw an Error to detect early an invalid `filters` key to avoid long lasting regressions.

File: frontend/src/lib/utils/api-routes.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+interface ParamsThirdPartyUrlmodelId extends BaseParams {
+	model: thirdPartyUrlModel;
+	id: string;
+}
+
+export const urlmodelIdGET: RequestHandler<ParamsUrlmodelId> = async ({ fetch, params, url }) => {
+	const model = getModelInfo(params.model);
+	const endpoint = `${BASE_API_URL}/${model.endpointUrl ?? params.model}/${params.id}/${
+		url.searchParams ? '?' + url.searchParams.toString() : ''
+	}`;
```
[the remark this answers is not in the document]
Comment (monsieurswag) [reply frontend/src/lib/utils/api-routes.ts:39]: This error comes from copy-pasting the current codebase code (it's not my fault :( ).

**Comment by ab-smith** [comment]
The number of files affected is concerning and we need some tests first before doing anything here to avoid another wave of regressions (batch mode, duplicate, pick existing from nested tables and more)

**Comment by monsieurswag** [comment]
> The number of files affected is concerning and we need some tests first before doing anything here to avoid another wave of regressions (batch mode, duplicate, pick existing from nested tables and more)

In order for these newly created `+server.ts` file to provoke any regression, they have to be queried by the application.
But they are not queried by the application, and the proof of it is that if it was the case we would already have regression as before this PR these sveltekit endpoints where unreachable because of route conflicts.

Apart from that there are ~10 modified `+server.ts` they have all been modified in the exact same way to correct a ternary operator which used the wrong way of checking if an `URLSearchParams` was empty or not.

The generic routes `+server.ts` in themselves weren't that much modified (their request handler have just been moved to `api-routes.ts` without their logic being truly modified).

The newly introduced test file is "safe" from regression (as it only runs in the CI context).

So if we sum everything up the risk of regression is really not that high despite the high quantity of newly created + modified `+server.ts` files.

**Comment by ab-smith** [comment]
Let’s have some tests first anyway

**Comment by ab-smith** [comment]
Closing the PR but leaving the branch until it's ready.
